### PR TITLE
tools : Add script to check coding rules for .c and .h files.

### PR DESCRIPTION
**Pull request by Kheprys**
* Script based on uncrustify tools.
	* Use the same uncrustify configuration as RIOT-OS.

Signed-off-by: Stephen Clymans <stephen@clymans.fr>

### CODE REVIEW

File: tools/check-codingrules.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#version         :1
+#usage           :./check-codingrules.sh
+#==============================================================================
+
+# WARNING : This script must to be in GITROOT/tools directory, it will not work if you copy it elsewhere.
+
+# Bash code to get the real dir of the script, even if it's called from another path.
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" >/dev/null && pwd )"
```
Comment (gdoffe) [inline tools/check-codingrules.sh]: Use 4 spaces, no tabs. This comment can be applied for the entire file.

File: tools/check-codingrules.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		echo "That's all folks !"
+		exit 0
+	fi
+
+	read -n 1 -p "Launch meld to check diff ? (y/N) " answer
+	[ -z "$answer" ] && answer="N"
+	if [ ${answer^^} = "Y" ]; then
+	echo -e "\n"
+		if [ -e /usr/bin/meld ]; then
+			meld "$DIR/../" "$UNCRUSTIFY_OUTPUT_DIR" &
```
Comment (gdoffe) [inline tools/check-codingrules.sh]: Why running it in background ? Script should wait end of meld.

**Comment by Kheprys** [comment]
Requested changes has been taken into account. Can be merged in master.

**Comment by gdoffe** [comment]
No upper case nor punctuation in commit title:

> tools : add script to check coding rules for .c and .h files

**Pull request merged**